dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java

```diff
@@ -102,31 +105,49 @@ public FileAssetAPIImpl(
         this.contentletCache = contentletCache;
     }
 
+	/**
+	 * Gets all the files,that the user has permission, under the specified parent.
+	 *
+	 * @param parent Parent where the files lives could be a folder or a Host
+	 * @param sortBy sort
+	 * @param working boolean include working files if true
+	 * @param archived boolean include archived files if true
+	 * @param user user
+	 * @param respectFrontendRoles respect frontend roles
+	 * @return list of fileAssets
+	 */
+	@Override
+	@CloseDBIfOpened
+	public List<FileAsset> findFileAssetsByParentable(final Parentable parent,
```

Comment:
This new public method doesn't have tests

Reply:
done